src/emulation_core/mips/datapath.rs

```diff
@@ -951,14 +957,12 @@ impl MipsDatapath {
         // Set the result.
         self.state.alu_result = match self.signals.alu_control {
             AluControl::Addition => input1.wrapping_add(input2),
-            AluControl::AddWithNoWriteOnOverFlow => {
-                // Ugly hack to make addi and daddi work correctly
+            AluControl::AddWithNoWriteOnOverflow => {
                 if let RegWidth::Word = self.signals.reg_width {
-                    let i1 = input1 as u32;
-                    let i2 = input2 as u32;
-                    let sum = i1.overflowing_add(i2);
+                    let input1 = input1 as u32;
```

Comment:
We probably should worry about 64bit vs 32bit overflow

Reply:
Yeah, overflow is a whole thing on it's own we need to discuss and fix. There are many inconsistencies on how overflow is being handled, but I'm not mentally ready to address it until this PR is done being worked on

Reply:
Overflow handling is an inconsistency marked in documentation. Further discussion needed.